main.cpp

```diff
@@ -39,29 +39,427 @@ Purpose:  This project continues developing Project3.
+        float numPumpkinSeeds = 5.5f;
+        float numLinseed = 2.3f;
+        bool bioIngredients = 1;
+
+        void mixIngredients();
+        float getWeight(float flourUsed, float numPumpkinSeeds, float numLinseed);
+        bool burn(int bakingTime);
+    };
+
+    void bakeBread(RyeBread brd);
+    float cakeSell(std::string cakeName);
+    void smellGreat();
+    int minuteTimer(int timeAmount);
+
+    RyeBread breadToSell;
+};
+
+Bakery::Bakery() : flourAmount(22.7), numCake(10), numBread(38), numMoney(8392.21f), maxOvenTemp(350), minutesLeft(0), name("bakery")
+{
+    std::cout << "Bakery constructed." << "\n";
+}
+
+Bakery::~Bakery()
+{
+    std::cout << "Bakery destructed." << "\n";
+}
+
+Bakery::RyeBread::RyeBread()
+{
+    std::cout << "Nested type RyeBread constructed." << "\n";
+}
+
+Bakery::RyeBread::~RyeBread()
+{
+    std::cout << "Nested type RyeBread destructed." << "\n";
+}
+
+void Bakery::bakeBread(RyeBread brd)
+{
+    std::cout << "Flour used: " << flourAmount << std::endl << "Timer set to: " << brd.bakingTime << std::endl << "Oven set to: " << maxOvenTemp << std::endl;
+}
+
+float Bakery::cakeSell(std::string cakeName)
+{
+    if (cakeName == "Sweet Cake")
+    {
+        std::cout << "Product sold: " << cakeName << std::endl;
+        return 9430.f;
+    }
+
+    std::cout << "Product unavailable" << std::endl;
+    return 0.f;
+}
+
+void Bakery::smellGreat()
+{
+    std::cout << "Smelling great!" << std::endl;
+}
+
+int Bakery::minuteTimer(int timeAmount)
+{
+    minutesLeft = timeAmount;
+    while (minutesLeft >= 0)
+    {
+        if (minutesLeft > 1)
+        {
+            std::cout << "Baking, time left: " << minutesLeft << " minutes.\n";
+        }
+        else
+        {
+            if (minutesLeft > 0)
+            {
+                std::cout << "Baking, time left: " << minutesLeft << " minute.\n";
+            }
+            else
+            {
+                std::cout << "Time left: " << minutesLeft << " minutes. Baking completed.\n";
+                return minutesLeft;
+            }
+        }
+        --minutesLeft;        
+    }
+    std::cout << "Invalid input, please insert positive value. Setting timer to 0.\n";
+    return minutesLeft = 0;
+}
+
 /*
  copied UDT 3:
  */
 
+struct House
+{
+    House();
+    ~House();
+    int numWindows;
+    float livingRoomSize;
+    float roomHeight;
+    int numBathRooms;
+    float totalFloorSize;
+    int roomsToClean;
+    std::string name;
+
+    struct Bathroom
+    {
+        Bathroom();
+        ~Bathroom();
+        bool shower = true;
+        bool bathtub = false;
+        int numWindows = 1;
+        float mirrorSurface = 1.5f;
+        int numTowels = 3;
+
+        void getSteamy();
+        void turnWaterOn();
+        bool towelsState();
+    };
+
+    void provideShelter();
+    void provideRest(float sleepQuality);
+    bool getDirty(int numPeople, int numAnimals);
+    int cleaningProcedure(int numDirtyRooms, int secondsPerRoom);
```

Comment:
`cleaningProcedure` is a function name that does not contain an action (verb). 
Add a verb to this function name.
- [ ] revise